src/crush/CrushCompiler.cc

```diff
@@ -369,6 +369,12 @@ int CrushCompiler::decompile(ostream &out)
     case CRUSH_RULE_TYPE_ERASURE:
       out << "\ttype erasure\n";
       break;
+    case CRUSH_RULE_TYPE_MSR_FIRSTN:
+      out << "\ttype msr_firstn\n";
+      break;
+    case CRUSH_RULE_TYPE_MSR_INDEP:
+      out << "\ttype msr_indep\n";
+      break;
     default:
       out << "\ttype " << crush.get_rule_type(i) << "\n";
```

Comment:
I don't think so.  The harm would have been done when the encoding was created originally.  Allowing it to be decompiled this way rather than asserting would be more useful in debugging.  The resulting output wouldn't recompile as it would fail the assert you mentioned, so I don't see any harm in letting it be decompiled like this.